src/internal/message/message.go

```diff
@@ -46,10 +51,36 @@ func init() {
 	}
 
 	pterm.DefaultProgressbar.MaxWidth = 85
+
+	// Prepend the log filename with a timestampe
+	ts := time.Now().Format("2006-01-02-15-04-05")
+
+	// Try to create a temp log file
+	if logFile, err = os.CreateTemp("", fmt.Sprintf("zarf-%s-*.log", ts)); err != nil {
+		pterm.SetDefaultOutput(os.Stderr)
+		Error(err, "Error saving a log file")
+	} else {
+		// Otherwise fallback to stderr
+		logStream := io.MultiWriter(os.Stderr, logFile)
+		pterm.SetDefaultOutput(logStream)
+		message := fmt.Sprintf("Saving log file to %s", logFile.Name())
+		Note(message)
+	}
 }
 
-func debugPrinter(offset int) *pterm.PrefixPrinter {
-	return pterm.Debug.WithShowLineNumber(logLevel > 2).WithLineNumberOffset(offset)
+func debugPrinter(offset int, a ...any) {
+	printer := pterm.Debug.WithShowLineNumber(logLevel > 2).WithLineNumberOffset(offset)
+	printer.Println(a...)
+
+	// Always write to the log file
+	if logFile != nil && !pterm.PrintDebugMessages {
+		pterm.Debug.
+			WithShowLineNumber(true).
+			WithLineNumberOffset(offset).
+			WithDebugger(false).
+			WithWriter(logFile).
+			Println(a...)
+	}
 }
 
 func errorPrinter(offset int) *pterm.PrefixPrinter {
```

Comment:
If we are going to change our Debug printer, should we also standardize our other printers so that they are consistent when called?

Reply:
This is just a work-around for the debug printer, because if you don't do this, it will never print the debug message to the log file.